airflow-core/src/airflow/triggers/deadline.py

```diff
@@ -49,26 +49,25 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
         from airflow.models.deadline import DeadlineCallbackState  # to avoid cyclic imports
 
         try:
-            callback = import_string(self.callback_path)
             yield TriggerEvent({PAYLOAD_STATUS_KEY: DeadlineCallbackState.RUNNING})
+            callback = import_string(self.callback_path)
```

Comment:
Any reason or impact for this swap around the yield?

Reply:
Nothing significant - I just did it so that if there was a failure while importing and someone was monitoring the state of the callback, it would go from queued to running and then failed instead of queued -> failed